src/cart/utils/memcheck-cart.supp

```diff
@@ -271,3 +271,89 @@
    fun:HG_Init_opt
    ...
 }
+{
+   crt_grp_lc_uri_insert
+   Memcheck:Leak
```

Comment:
I'll need some convincing that whitelisting leaks is an improvement, if we have residual leaks then we should be aiming to resolve them and only suppressing them whilst that work is ongoing, but this PR seems to only add suppressions for leaks with no code changes so doesn't make a lot of sense to me.

Reply:
So change [CART-975](https://daosio.atlassian.net/browse/CART-975) to documenting, resolving, or reporting these leaks to Mercury or libfabric?  

We might want to _not_ repeatedly report the [681 memory leaks](https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-6743/1/) while we're resolving them in CI.  (But since we're not currently reporting leaks as errors, that's moot while [CART-977](https://daosio.atlassian.net/browse/CART-977) is open.)

Reply:
the crt_grp_lc_uri_insert leak seems to be related to https://daosio.atlassian.net/browse/DAOS-8212

Reply:
I'm not sure I follow, and I don't see 681 leaks reported in the above link.

You can configure the valgrind plugin with various thresholds, but that we've done in the past is to set all thresholds to zero, initially suppress any warnings and then gone through the suppression file(s) either resolding or reporting leaks upstream as appropriate.  I'm not sure where the Cart checks are in this process but I'm not aware of master failing with reported leaks and this PR only touches the suppression file hence my confusion.

Specifically with respect to CART-975 it is not adequate to suppress leaks and close the ticket, no.

Reply:
> I'm not sure I follow, and I don't see 681 leaks reported in the above link.

I view the leaks by downloading [the valgrind memcheck xml files](https://build.hpdd.intel.com/job/daos-stack/job/daos/view/change-requests/job/PR-6743/1/artifact/Functional%20on%20CentOS%207%20with%20Valgrind/cart/*zip*/cart.zip).

> You can configure the valgrind plugin with various thresholds, but that we've done in the past is to set all thresholds to zero, initially suppress any warnings and then gone through the suppression file(s) either resolding or reporting leaks upstream as appropriate. I'm not sure where the Cart checks are in this process but I'm not aware of master failing with reported leaks and this PR only touches the suppression file hence my confusion.

[CART-972](https://daosio.atlassian.net/browse/CART-972) is open to use the Jenkins Valgrind plugin.